osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -164,7 +164,7 @@ private void load(AudioManager audio)
                                                     new DrawableMatchRoom(Room, allowEdit)
                                                     {
                                                         OnEdit = () => settingsOverlay.Show(),
-                                                        SelectedItem = { BindTarget = SelectedItem }
+                                                        SelectedItem = SelectedItem
```

Comment:
I'm really hesitant about this one. Assigning directly to a bindable always feels wrong. Reading back on the [previous PR](https://github.com/ppy/osu/pull/30592), the rationale seems to be that "this means we won't forget to set it"? I dunno how valid that is. Would keeping the old `BindTarget` method still be feasible?

And yes, we use `BindableWithCurrent` elsewhere like this. The difference I see is that in all those usages (to my knowledge), the property is named `Current`, which signifies that it has the special underlying behaviour.

In addition, we're passing this multiple levels down (see `MultiplayerMatchFooter` / `MatchStartControl` as one example). We usually tend to use DI in such a case. Could we create a `MultiplayerState` cached class for containing things like this as an alternative? Potentially also containing `Room` which seems to be passed everywhere?

Reply:
I think this is one of the most important parts of this series of PRs...

> the rationale seems to be that "this means we won't forget to set it"?

It enforces a contract, in the same way a ctor enforces a contract. `BindTarget`/`BindTo` and all that work, but it must _always_ be assigned. At compile time it says that if you haven't assigned this you're _definitely_ doing something wrong, so think about what you want to do now.  
In previous code this would've been an ambient non-nullable resolved bindable that enforced the same contract at run-time, which imo is very hidden, especially for something that potentially has multiple sources (I've only applied this to `SelectedItem`, which is exactly this case).

> the property is named `Current`, which signifies that it has the special underlying behaviour.

This is not significance to me because I made no effort to use or reproduce `IHasCurrentValue<>`. I only wanted to enforce the contract.

> Could we create a `MultiplayerState` cached class for containing things like this as an alternative?

To be honest, if you want that then this series of PRs has no need to exist. Because, the natural follow up is - if we BDL `Room`, then we might as well keep every property in `Room` also bindable. Most of the complexity of this PR goes away I believe.

It's a different take on things. I can't say whether that would be a bad idea.

Reply:
As discussed IRL, my proposed direction is likely not where we want to go. I can agree that we should avoid caching "states" to DI, especially when those states are not singleton (smoogi mentioned that if we have states here, there will need to be multiple layers to handle lobby/room).

So let's leave things how they are for now.